packages/cli/src/auth/index.ts

```diff
@@ -1,86 +1,243 @@
+    return creds.token;
   } else {
-    const apiKey = await createSession(cache);
-    return apiKey;
+    if (showLogs) {
+      console.log(
+        chalk.yellow(
+          "We did not find a session, please connect your wallet through our dashboard to continue",
+        ),
+      );
+    }
+    await getOrGenerateLocalWallet(credsConfigPath, cliWalletPath);
+    const authToken = await authenticateUser({ browser: true, credsConfigPath, cliWalletPath });
+    if (!authToken) {
+      throw new Error("Failed to login");
+    }
+    return authToken;
   }
 }
 
-export async function logoutUser(cache: Cache) {
+export async function logoutUser(credsConfigPath: string) {
   try {
-    cache.remove("api-secret-key");
+    const dirExists = fs.existsSync(credsConfigPath);
+    if (!dirExists) {
+      return;
+    }
+    const configJson = fs.readFileSync(credsConfigPath, "utf-8");
+    const parsedJson = JSON.parse(configJson);
+    parsedJson.password = "";
+    parsedJson.token = "";
+    fs.writeFileSync(credsConfigPath, JSON.stringify(parsedJson), "utf-8");
     console.log(chalk.green("You have been logged out"));
   } catch (error) {
     console.log(chalk.red("Something went wrong", error));
   }
 }
 
-export function getSession(cache: Cache) {
+export function getSession(credsConfigPath: string) {
   try {
-    const apiKey: CacheEntry = cache.get("api-secret-key");
-    return apiKey.value;
+    const parsedConfigJson = JSON.parse(fs.readFileSync(credsConfigPath, "utf8"));
+    return parsedConfigJson;
   } catch (error) {
     console.log(error);
   }
 }
 
-export async function createSession(cache: Cache) {
-  try {
-    const response = await prompts({
-      type: "invisible",
-      name: "apiSecretKey",
-      message: `Please enter your API secret key, you can find or create it on ${chalk.blue(
-        "https://thirdweb.com/create-api-key",
-      )}`,
+type LoginProps = {
+  browser: boolean;
+  credsConfigPath: string;
+  cliWalletPath: string;
+};
+
+export const authenticateUser = async (
+  props: LoginProps = {
+    browser: true,
+    credsConfigPath: "",
+    cliWalletPath: ""
+  },
+) => {
+  const { credsConfigPath, cliWalletPath } = props;
+
+  // Get or generate a localwallet.
+  const wallet = await getOrGenerateLocalWallet(credsConfigPath, cliWalletPath);
+  const walletAddress = await wallet.getAddress();
+  const auth = new ThirdwebAuth(wallet, "localhost:3000");
+
+  // Generate the login payload to pass to the dashboard.
+  const loggedIn = await auth.login({
+    domain: "localhost:3000",
+    address: walletAddress,
+  });
+
+  // In this case the state is the loggedIn object.
+  const ourState = encodeURIComponent(JSON.stringify(loggedIn));
+  const urlToOpen =
+    // `https://thirdweb.com/cli/login?from=cli&#${ourState}`;
+    // `https://thirdweb-www-git-mariano-api-keys-sign-in.thirdweb-preview.com/cli/login?from=cli&#${ourState}`;
+    `http://localhost:3000/cli/login?from=cli&#${ourState}`;
```

Comment:
don't need `from=cli` if you already have `/cli` as the path, right? should be clear where we're from at that point

either:
`/login/cli?payload=<the payload>&state=<the separate state value>`
or:
`/login?from=cli&payload=<the payload>&state<the separate state value>`